### DISCUSSION TITLE
igx-grid call (cellEdit) event on zero changes

### DISCUSSION THREAD

**Discussion by kevalgw** [IgniteUI/igniteui-angular#12155] [Q&A] [answered]
Hi,

In my igx-grid for certain validation i want to call (cellEdit) even if no changes are done. Is it possible to do? 
Please respond

**Comment by ddaribo** [comment, marked as answer]
Hi @kevalgw,

As per our API documenting the `cellEdit` event [here](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#cellEdit), it is fired _after_ editing is completed, when the cell is exiting edit mode. Thus it cannot really be fired without an edit. 

If you would like to execute some logic just before editing starts, I can suggest handling the [cellEditEnter](https://www.infragistics.com/products/ignite-ui-angular/docs/typescript/latest/classes/IgxGridComponent.html#cellEditEnter) event, which is emitted when cell enters edit mode.

Since you mention validation, I believe you will also find our new Grid Editing and Validation features useful. You can read more about them in [this topic](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/validation) in our documentation. This was introduced in Ignite UI for Angular version 14.1.0, so in order to take advantage of them your Ignite UI version has to be 14.1.0 or later.

**Reply by kevalgw** [reply]
Hi @ddaribo 

Thanks for the response.
